File: CS/SpreadsheetControl_API_Part03/CodeExamples/FormControlActions.cs
using DevExpress.Spreadsheet;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SpreadsheetControl_API_Part03.CodeExamples
{
    class FormControlActions
    {
        static void CreateFormControls(IWorkbook workbook)
        {
            #region #CreateFormControls
            var formControls = workbook.Worksheets[0].FormControls;

            // Create a button form control:
            var buttonCellRange = workbook.Worksheets[0].Range["B2:C2"];
            var buttonFormControl = formControls.AddButton(buttonCellRange);
            buttonFormControl.PlainText = "Click Here";

            // Create a list box form control:
            var comboCellRange = workbook.Worksheets[0].Range["B4:C4"];
            var comboBoxControl = formControls.AddComboBox(comboCellRange);
            comboBoxControl.DropDownLines = 3;
            comboBoxControl.SourceRange = workbook.Worksheets[0].Range["E2:E6"];
            comboBoxControl.SelectedIndex = 1;

            // Create a check box form control:
            var checkRange = workbook.Worksheets[0].Range["D5:E5"];
            var checkBoxControl = formControls.AddCheckBox(checkRange);
            checkBoxControl.CheckState = FormControlCheckState.Checked;
            checkBoxControl.PlainText = "Reviewed";
            #endregion #CreateFormControls
        }

        static void EditFormControls(IWorkbook workbook)
        {
            #region #EditFormControls
            workbook.LoadDocument("Documents\\FormControls.xlsx");

            var formControls = workbook.Worksheets[0].FormControls;

            foreach (FormControl formControl in formControls)
            {
                formControl.PrintObject = false;
            }
            #endregion #EditFormControls
        }
    }
}

File: VB/SpreadsheetControl_API_Part03/CodeExamples/FormControlActions.vb
Imports DevExpress.Spreadsheet
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace SpreadsheetControl_API_Part03.CodeExamples

    Friend Class FormControlActions

        Private Shared Sub CreateFormControls(ByVal workbook As DevExpress.Spreadsheet.IWorkbook)
#Region "#CreateFormControls"
            Dim formControls = workbook.Worksheets(CInt((0))).FormControls
            ' Create a button form control:
            Dim buttonCellRange = workbook.Worksheets(CInt((0))).Range("B2:C2")
            Dim buttonFormControl = formControls.AddButton(buttonCellRange)
            buttonFormControl.PlainText = "Click Here"
            ' Create a list box form control:
            Dim comboCellRange = workbook.Worksheets(CInt((0))).Range("B4:C4")
            Dim comboBoxControl = formControls.AddComboBox(comboCellRange)
            comboBoxControl.DropDownLines = 3
            comboBoxControl.SourceRange = workbook.Worksheets(CInt((0))).Range("E2:E6")
            comboBoxControl.SelectedIndex = 1
            ' Create a check box form control:
            Dim checkRange = workbook.Worksheets(CInt((0))).Range("D5:E5")
            Dim checkBoxControl = formControls.AddCheckBox(checkRange)
            checkBoxControl.CheckState = DevExpress.Spreadsheet.FormControlCheckState.Checked
            checkBoxControl.PlainText = "Reviewed"
#End Region  ' #CreateFormControls
        End Sub

        Private Shared Sub EditFormControls(ByVal workbook As DevExpress.Spreadsheet.IWorkbook)
#Region "#EditFormControls"
            workbook.LoadDocument("Documents\FormControls.xlsx")
            Dim formControls = workbook.Worksheets(CInt((0))).FormControls
            For Each formControl As DevExpress.Spreadsheet.FormControl In formControls
                formControl.PrintObject = False
            Next
#End Region  ' #EditFormControls
        End Sub
    End Class
End Namespace
